wagtail_transfer/wagtail_hooks.py

```diff
@@ -25,9 +24,21 @@ def register_admin_urls():
 
 class WagtailTransferMenuItem(MenuItem):
     def is_shown(self, request):
-        return bool(getattr(settings, 'WAGTAILTRANSFER_SOURCES', None))
+        return all(
+            [
+                request.user.has_perm("wagtailcore.wagtailtransfer_can_import"),
+                bool(getattr(settings, "WAGTAILTRANSFER_SOURCES", None)),
```

Comment:
**Todo**: I'll reverse these, so that the settings attribute check can shortcut the database query

Reply:
I think having it in a list rather than using eg a generator means it'll evaluate both anyway, but it might be wrong - but otherwise, you might have to rewrite it

Reply:
From a well-known code question and answer site, I was pointed to the example 'equivalent' implementation of `any` in the Python docs: https://docs.python.org/3/library/functions.html#all, where the short-circuit is implied, and a pull request which guarantees the short-circuiting behaviour via library unit tests, now at https://hg.python.org/cpython/file/tip/Lib/test/test_builtin.py#l187. Thanks for prompting me to check that.

Reply:
Nice! Sorry about that, I'd always thought a list would get fully evaluated before `all` ran - great to learn that shortcut is in there.

Reply:
> Sorry about that
My thanks were genuine: it was good to check my assumptions and learn something too.

Reply:
Done in a830b00.